### ISSUE TITLE
[Bug] Wrong lazy relation value as parameter on ManyToOne relations

### ISSUE THREAD

**Issue by luchillo17** [typeorm/typeorm#824]
So i'm debugging why it uses the primary key value of the `Many` side instead of the one in the relation defined by `@JoinColumn`.

Here's how it looks:
```ts
@Entity('a_asset', {
  skipSync: true,
  orderBy: {
    id: 'DESC',
  },
} as any)
export class Asset {
  @PrimaryColumn('int', {
    name: 'a_asset_id',
  })
  id: number

  // Relations
  @ManyToOne(
    (type) => Organization,
    (organization) => organization.assets,
  )
  @JoinColumn({
    name: 'ad_org_id',
  })
  organization: Promise<Organization>
}
```
And the organization one:
```ts
@Entity('ad_org', {
  skipSync: true,
  orderBy: {
    id: 'DESC',
  },
} as any)
export class Organization {
  @PrimaryColumn('int', {
    name: 'ad_org_id',
  })
  id: number

  // Relations
  @OneToMany(
    (type) => Asset,
    (asset) => asset.organization,
  )
  assets: Promise<Asset[]>
}
```
It seems to me that the value of `@JoinColumn` property is not being saved to the `Asset` result object before using it for the `Lazy-Relation`, so the Lazy relation by default tries with the `primaryKey`.

**Comment by makamekm** [comment]
What version do you use?

**Comment by luchillo17** [comment]
Directly from GitHub.

**Comment by luchillo17** [comment]
Ok so i'm logging the entity before it get's passed to the `LazyRelationsWrapper` and this shows:
```ts
let result = [ Asset {
    id: '1015712',
    name: '582ADJ',
    value: '582ADJ',
    versionno: '2018',
    organizationId: '0' } ]
```
As you can see there's `organizationId`, however when i go to the wrapper, when i go into calling the `getEntityValue` function on the join column, i see this:
![image](https://user-images.githubusercontent.com/9698639/30169108-0c43bc7a-93b1-11e7-9af0-58a4a238b9ee.png)
Take note that there's an `entityMetadata`, but no `embeddedMetadata` nor `relationMetadata` which are the ones the `getEntityValue` asks for, in such case it goes directly for the `id` field of entity instead of the relation one.

**Comment by luchillo17** [comment]
Check https://github.com/typeorm/typeorm/issues/826 to see my situation regarding testing, skipping schema tests meanwhile.

I'm making the testing for all kind of Lazy relations, from basic to named tables, and from basic to named PK columns, as well as all or most combination of such, got errors after just adding the relations in entity, without touching the tests yet:
![image](https://user-images.githubusercontent.com/9698639/30235819-3978f6ba-94d3-11e7-98a1-a0d40f1b5f91.png)
That's the error issue, check the last `ON` clause before the `AND`, check out how left part of comparison has no quoting at all, that's one of the bugs my fix well... fixes.

**Comment by luchillo17** [comment]
@pleerock I've been testing and seems like the error above breaks in MySql, dunno SqLite since i have it always off, but all other databases works ok, could it be the naming strategy with named tables?, or the MySql driver?

**Comment by pleerock** [comment]
most of problems with lazy relations must be inside `LazyRelationsWrapper` class. If you cant find the way to fix them just push your tests Ill mege them and try to fix them by myself

**Comment by luchillo17** [comment]
Ok however as of now the only thing i did was adding relations for named tables, named PK columns, and a combination of both to the entities, so far haven't changed the tests.

Here it is: https://github.com/typeorm/typeorm/pull/853

**Comment by luchillo17** [comment]
Closed with merge in #853.
